tests/MockApi.php

```diff
@@ -0,0 +1,154 @@
+<?php
+namespace Auth0\Tests;
+
+use Auth0\SDK\API\Management;
+use Auth0\SDK\API\Management\Blacklists as BL;
+use Auth0\SDK\API\Management\Clients as Cl;
+use Auth0\SDK\API\Management\ClientGrants as CG;
+use Auth0\SDK\API\Management\Connections as Co;
+use Auth0\SDK\API\Management\DeviceCredentials as DC;
+use Auth0\SDK\API\Management\Emails as Em;
+use Auth0\SDK\API\Management\EmailTemplates as ET;
+use Auth0\SDK\API\Management\Jobs as J;
+use Auth0\SDK\API\Management\Logs as L;
+use Auth0\SDK\API\Management\ResourceServers as RS;
+use Auth0\SDK\API\Management\Rules as R;
+use Auth0\SDK\API\Management\Stats as S;
+use Auth0\SDK\API\Management\Tenants as Te;
+use Auth0\SDK\API\Management\Tickets as Ti;
+use Auth0\SDK\API\Management\UserBlocks as UB;
+use Auth0\SDK\API\Management\Users as U;
+use Auth0\SDK\API\Management\UsersByEmail as UE;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * Class MockApi
+ *
+ * @package Auth0\Tests
+ */
+class MockApi
+{
+
+    /**
+     * Base API URL for the mock API.
+     */
+    const API_BASE_URL = 'https://api.test.local/api/v2/';
+
+    /**
+     * Guzzle request history container for mock API.
+     *
+     * @var array
+     */
+    protected $requestHistory = [];
+
+    /**
+     * Management API object.
+     *
+     * @var Management
+     */
+    protected $client;
+
+    /**
+     * Management API endpoint name.
+     *
+     * @var string
+     */
+    protected $endpoint;
+
+    /**
+     * MockApi constructor.
+     *
+     * @param string $endpoint  Endpoint property name to call.
+     * @param array  $responses Responses to be loaded, an array of Response objects.
+     */
+    public function __construct($endpoint, array $responses = [])
+    {
+        $guzzleOptions = [];
+        if (count( $responses )) {
+            $mock    = new MockHandler($responses);
+            $handler = HandlerStack::create($mock);
+            $handler->push( Middleware::history($this->requestHistory) );
+            $guzzleOptions['handler'] = $handler;
+        }
+
+        $this->client   = new Management('__api_token__', 'api.test.local', $guzzleOptions);
+        $this->endpoint = $endpoint;
+    }
+
+    /**
+     * Return the endpoint being used.
+     *
+     * @return BL | Cl | CG | Co | DC | Em | ET | J | L | R | RS | S | Te | Ti | UB | U | UE
```

Comment:
what are these?

Reply:
Gone now 👍

Reply:
I think `@return` line can be multiline. Anyway, never minify classes' names. People shouldn't be trying to guess what you mean. Is fine if you want to add them back instead of removing them, just use the full name. 👌

Reply:
> I think @return line can be multiline.

IDE won't parse it. Trust me, I don't like this either but it retained the functionality needed and is "legal" according to PHP :)

Either way, not needed now.